metadata-ingestion/src/datahub/ingestion/source/bigquery_v2/lineage.py

```diff
@@ -176,12 +170,12 @@ def lineage_via_catalog_lineage_api(
             datasets = list(bigquery_client.list_datasets(project_id))
             project_tables = []
             for dataset in datasets:
-                # Enables only tables where type is TABLE (removes VIEWS)
+                # Enables only tables where type is TABLE, VIEW or MATERIALIZED_VIEW (not EXTERNAL)
                 project_tables.extend(
                     [
                         table
                         for table in bigquery_client.list_tables(dataset.dataset_id)
-                        if table.table_type == "TABLE"
+                        if table.table_type in ["TABLE", "VIEW", "MATERIALIZED_VIEW"]
```

Comment:
I know this wasn't previously the case, but can you use `BigqueryTableType` over the string constants here. Also, are you sure the correct table type is `MATERIALIZED_VIEW` not `MATERIALIZED VIEW` (as per https://cloud.google.com/bigquery/docs/information-schema-tables#schema)? I tried searching the data catalog lineage docs but couldn't find an answer.

Reply:
Hi @asikowitz, it's here https://cloud.google.com/python/docs/reference/bigquery/latest/google.cloud.bigquery.table.Table#google_cloud_bigquery_table_Table_table_type

<img width="592" alt="image" src="https://github.com/datahub-project/datahub/assets/6565443/d18a9492-dd85-4acf-8c07-0eab226d61e1">

I've checked the BigQuery SDK, and I didn't find the ENUM with the table types.

Reply:
Can we proceed with the strings? Otherwise, I'd need to open a PR to the BigQuery Python repo to create something to use.